### PR TITLE
ci: add pr workflow to check pr title format

### PR DESCRIPTION
**Pull request by asraa**
* ci: adds a ci workflow on PRs to check the PR title format against regexes that follow conventional commits

Part of https://github.com/slsa-framework/slsa-verifier/issues/371

Signed-off-by: Asra Ali <asraa@google.com>

### CODE REVIEW

File: .github/pr-title-checker-config.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+
+{
+    "LABEL": {
+      "name": "title needs adjustment",
+      "color": "EEEEEE"
+    },
+    "CHECKS": {
+      "prefixes": [
+        "chore: "      ],
```
Comment (ianlewis) [inline .github/pr-title-checker-config.json:9]: nit this file is formatted a bit strangely.

Comment (ianlewis) [inline .github/pr-title-checker-config.json:9]: nit: 'chore' seems to be handled by the regex?

**Review by ianlewis — APPROVED** [review]
We could use a similar config in slsa-github-generator's repo too.

File: .github/pr-title-checker-config.json (hunk 1 above)
Comment (asraa) [reply .github/pr-title-checker-config.json]: removing that cleans it up a bit.

**Pull request merged**